.github/workflows/workflow.yml

```diff
@@ -302,18 +337,65 @@ jobs:
           ${{ matrix.integration_test }}_integration_tests
         )
 
+  kind_integration_github:
+    strategy:
+      matrix:
+        integration_test: [deep, upgrade, helm, custom_domain, external_issuer]
+    needs: [docker_build]
+    name: GitHub int. tests (${{ matrix.integration_test }})
+    if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Download image archives
+      uses: actions/download-artifact@v1
+      with:
+        name: image-archives
+    - name: Try to load cached Go modules
+      uses: actions/cache@v1
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
```

Comment:
similar question to above, what's this caching for? is it intended to speed up the `Run integration tests` below?

Reply:
Here loading cached Go modules is helpful for running integration tests. [Here](https://github.com/kleimkuhler/linkerd2/runs/424047380?check_suite_focus=true) is a workflow that was able to load a cache first, and [here](https://github.com/kleimkuhler/linkerd2/runs/425936894?check_suite_focus=true) is a workflow that did not. Notice how the `Run integration tests` step had to download all the Go modules first before running.